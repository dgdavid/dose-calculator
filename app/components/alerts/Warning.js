import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text } from 'react-native';

import colors from '../../config/colors';

const Warning = ({ visible, text }) => {
  if (!visible) {
    return null;
  }

  return (
    <Fragment>
      <Text style={styles.emoji}>{'\u26A0\uFE0F'}</Text>
      <Text style={styles.text}>{text}</Text>
    </Fragment>
  );
};

Warning.propTypes = {
  visible: PropTypes.bool,
  text: PropTypes.string.isRequired,
};

Warning.defaultProps = {
  visible: false,
};

const styles = StyleSheet.create({
  emoji: {
    fontSize: 15,
    alignSelf: 'center',
  },

  text: {
    color: colors.materialDeepOrange500,
    fontSize: 10,
    alignSelf: 'center',
  },
});

export default Warning;
