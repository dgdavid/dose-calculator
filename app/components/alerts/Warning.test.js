import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Warning from './Warning';

const text = 'Example text for test `Warning`';

test('renders correctly when visible is `true`', () => {
  const tree = renderer.create(<Warning text={text} visible />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('rendering nothing when visible is `false`', () => {
  const tree = renderer.create(<Warning text={text} visible={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('rendering nothing when visible is not given', () => {
  const tree = renderer.create(<Warning text={text} />).toJSON();

  expect(tree).toMatchSnapshot();
});
