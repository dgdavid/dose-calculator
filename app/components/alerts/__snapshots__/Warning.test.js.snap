// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering nothing when visible is \`false\` 1`] = `null`;

exports[`rendering nothing when visible is not given 1`] = `null`;

exports[`renders correctly when visible is \`true\` 1`] = `
Array [
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 15,
      }
    }
  >
    ⚠️
  </Text>,
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF5722",
        "fontSize": 10,
      }
    }
  >
    Example text for test \`Warning\`
  </Text>,
]
`;
